From a29bca1adf776ea90b2b8fafd33c6acb8842c1a9 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 18 Jul 2022 10:16:23 +0300
Subject: [PATCH] [SW-94864] rom: rename boot info apis to export

Boot info apis in ROM do not support merge functionality. These apis
can be used in other FW components. Hence the apis are renamed in
order to export the same.

Change-Id: Ie106af18167ac7cf348296d2452fc20eff17cbfc
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 0aae899..7c74788 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -654,9 +654,9 @@
 	 * re-validation of the spi images
 	 */
 	if (!arcboot_rom_global.validation_pass) {
-		boot_info_set(HL_BOOT_INFO_ROM_SECURITY_DATA, &arcboot_rom_global.sec_data,
+		boot_info_set_no_merge(HL_BOOT_INFO_ROM_SECURITY_DATA, &arcboot_rom_global.sec_data,
 				sizeof(arcboot_rom_global.sec_data));
-		boot_info_finalize();
+		boot_info_finalize_no_merge();
 	}
 }
 
diff --git a/common/boot_desc.c b/common/boot_desc.c
index a7ff9e4..acb8cc4 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -131,28 +131,28 @@
 skip_boot_desc_process:
 	if (img == HL_FW_BOOT_PTR_SPI) {
 		if (!arcboot_rom_global.validation_pass) {
-			if (boot_info_set(HL_BOOT_INFO_IMG_DESC_INFO, &p_img_ptr_ctx->desc_info,
+			if (boot_info_set_no_merge(HL_BOOT_INFO_IMG_DESC_INFO, &p_img_ptr_ctx->desc_info,
 			   sizeof(struct hl_fw_desc_info))) {
 				arcboot_rom_status_update(PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL);
 				return 0;
 			}
 
-			if (boot_info_set(HL_BOOT_INFO_IMG_DESC_DATA,
+			if (boot_info_set_no_merge(HL_BOOT_INFO_IMG_DESC_DATA,
 			   (uint8_t *)(p_img_ptr_ctx->dest_addr_arc + p_offset[0]),
 			   p_img_ptr_ctx->desc_info.desc_size * p_img_ptr_ctx->desc_info.count)) {
 				arcboot_rom_status_update(PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL);
 				return 0;
 			}
 
-			boot_info_finalize();
+			boot_info_finalize_no_merge();
 		}
-		if (boot_info_get(HL_BOOT_INFO_IMG_DESC_INFO, &p_img_ptr_ctx->desc_info,
+		if (boot_info_get_no_merge(HL_BOOT_INFO_IMG_DESC_INFO, &p_img_ptr_ctx->desc_info,
 		   sizeof(struct hl_fw_desc_info), &copy_sz)) {
 			arcboot_rom_status_update(PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL);
 			return 0;
 		}
 
-		if (boot_info_get(HL_BOOT_INFO_IMG_DESC_DATA,
+		if (boot_info_get_no_merge(HL_BOOT_INFO_IMG_DESC_DATA,
 		   (uint8_t *)(p_img_ptr_ctx->dest_addr_arc + BOOT_INFO_IMG_DESC_GET_OFFSET),
 		   p_img_ptr_ctx->desc_info.desc_size * p_img_ptr_ctx->desc_info.count, &copy_sz)) {
 			arcboot_rom_status_update(PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL);
diff --git a/common/boot_info.c b/common/boot_info.c
index 87bda09..02d4460 100644
--- a/common/boot_info.c
+++ b/common/boot_info.c
@@ -64,7 +64,7 @@
 	return memcmp(&crc32_res, &desc->crc32, CRC32_NUM_BYTES);
 }
 
-void boot_info_finalize(void)
+void boot_info_finalize_no_merge(void)
 {
 	struct hl_boot_info_desc *desc;
 	uint32_t offsets[MAX_INFO_DESCS];
@@ -132,7 +132,7 @@
 		dst[i] = boot_info.data[i];
 }
 
-int boot_info_set(uint32_t type, const void *data, uint32_t size)
+int boot_info_set_no_merge(uint32_t type, const void *data, uint32_t size)
 {
 	struct hl_boot_info_desc *desc;
 
@@ -178,7 +178,7 @@
 	memcpy(&d[i], &rem, rem_sz);
 }
 
-int boot_info_get(uint32_t type, void *data, uint32_t max_size, uint32_t *copy_size)
+int boot_info_get_no_merge(uint32_t type, void *data, uint32_t max_size, uint32_t *copy_size)
 {
 	struct hl_boot_info_ptrs ptrs;
 	struct hl_boot_info_desc desc;
diff --git a/common/boot_info.h b/common/boot_info.h
index 0eef30b..1d95da3 100644
--- a/common/boot_info.h
+++ b/common/boot_info.h
@@ -20,9 +20,9 @@
  */
 #define BOOT_INFO_IMG_DESC_GET_OFFSET	(128 * 1024) // 128KB offset
 
-void boot_info_finalize(void);
-int boot_info_set(uint32_t type, const void *data, uint32_t size);
-int boot_info_get(uint32_t type, void *data, uint32_t max_size, uint32_t *copy_size);
+void boot_info_finalize_no_merge(void);
+int boot_info_set_no_merge(uint32_t type, const void *data, uint32_t size);
+int boot_info_get_no_merge(uint32_t type, void *data, uint32_t max_size, uint32_t *copy_size);
 uint16_t hl_get_boot_info_ver(void);
 
 #endif /* _BOOT_INFO_H_ */
